trackintel/preprocessing/triplegs.py

```diff
@@ -1,6 +1,20 @@
+from shapely.geometry import LineString
+from simplification.cutil import simplify_coords#, simplify_coordsvw
+import copy
+import ast
 
-
-def smoothen_triplegs(self, triplegs, method='douglas-peucker'):
-    """Not implemented.
+def smoothen_triplegs(triplegs, method='douglas-peucker', epsilon = 1.0):
+    """reduces number of points while retaining structure of tripleg
+    Parameters
+    ----------
+    triplegs: shapely file
+        triplegs to be reduced
+    method: method used to smoothen
+        only one method available so far
+    epsilon: float
+        slack parameter, higher epsilon means removing more points
     """
-    pass
\ No newline at end of file
+    input_copy = copy.deepcopy(triplegs)
+    input_copy.geom = [LineString(ast.literal_eval(str(simplify_coords(input_copy.geom[i].coords, epsilon))))
+                       for i in range(len(input_copy.geom))]
```

Comment:
Why go through the AST? `simplify_coords` should already return a list of coordinate pairs (which can directly be passed to the LineString constructor)?
Also the last `.geom` is probably not needed.